src/vivarium_public_health/results/stratification.py

```diff
@@ -62,15 +62,14 @@ def register_stratifications(self, builder: Builder) -> None:
             is_vectorized=True,
             requires_columns=["current_time"],
         )
-        # TODO [MIC-4232]: simulants occasionally have event year of end_year_year+1 if the end time plus step size
-        #  lands in the next year. possible solution detailed in ticket
-        # builder.results.register_stratification(
-        #     "event_year",
-        #     [str(year) for year in range(self.start_year, self.end_year + 1)],
-        #     mapper=self.map_year,
-        #     is_vectorized=True,
-        #     requires_columns=["event_time"],
-        # )
+        builder.results.register_stratification(
```

Comment:
This is very similar to how we register current_year, above, except 
1. We extend the categories range by one year (since the last event of a sim is often after the sim's end)
2. We then exclude that last year since it's past the sim's end